src/tools/sfstools.sh

```diff
@@ -1,5 +1,9 @@
 #!/usr/bin/env bash
 
-tool=$(basename $0)
-
-${tool/saunafs/saunafs } "$@"
+tool="$(basename $0)"
+case "$tool" in
+  (lizardfs*) tool="${tool#lizardfs}" ;;
+  (mfs*) tool="${tool#mfs}" ;;
+  (saunafs*) tool="${tool#saunafs}" ;;
```

Comment:
It is not clear to me if this should also catch `sfs*`, see also: https://github.com/leil-io/saunafs/commit/64f3f27c1ee7550156192367cbcedeeaac00039d

Reply:
It actually should. `sfs` is what `mfs` used to stand for in the code base. `saunafs` is the renamed `lizardfs` tool, which is the binary this script will call. So I don't think there is even a need to add a switch between `saunafs` (`lizardfs`) and `sfs` (`mfs`) as calling `saunafs` directly won't invoke this script at all.

Could you also please remove the `mfs` and `lizardfs`? We don't want to conflict with either MooseFS or LizardFS (as we renamed everything to not conflict, these will not be an issue anymore).

Reply:
I warn you therefore, you should know that this changes the behavior of the script to strip `sfs` instead of `saunafs` as before. Do you have any release notes where this should be mentioned for package maintainers?

Reply:
> you should know that this changes the behavior of the script to strip sfs instead of saunafs as before

This must have been some kind of regression. It should have replaced mfs with lizardfs in LizardFS, but now I see that in LizardFS it's replacing lizardfs with lizardfs!

Digging deeper, the original commit for this looked like this: https://github.com/leil-io/saunafs/commit/23e3af5ac794c0cb790abefa712a6ec40eabd1d9#diff-61ffda4854ab1bec0eabeee8ef0cb7fb6ba6a7c41dd34845e97442dc7fcff00aR1

But this changed with this commit
https://github.com/leil-io/saunafs/commit/4684bb21667b126f14f4c11333e46af46488f1d8#diff-61ffda4854ab1bec0eabeee8ef0cb7fb6ba6a7c41dd34845e97442dc7fcff00aR5

I can only assume this was a mistake, since there's no reason to replace lizardfs with lizardfs, the mfs set of tools (now sfs) are simply this script file.

Why this was never caught I'll never know.

> Do you have any release notes where this should be mentioned for package maintainers?

We have a changelog that automatically fetches commits since previous stable release when we release, and we try to avoid stuff that breaks packaging in minor/patch versions (i.e 4.4.0, the middle and right number should not break anything, including packaging, however the left one could.). I don't believe this could affect package maintainers (unless they fixed this themselves), but perhaps for breaking changes, we could potentially use ReleaseNotes for that since that hasn't been updated since LizardFS (outside of accidentally renaming LizardFS in there to SaunaFS).

Reply:
there is a reason to replace `lizardfs` with `lizardfs ` (notice the extra space), and it's also exactly what the original script (in the patch this was ported from) did. Are you sure that you have no more references to `mfs` in the code base (which caused issues in LizardFS)?

Reply:
> there is a reason to replace `lizardfs` with `lizardfs ` (notice the extra space)

Ah I see now. But it still doesn't explain why the match was changed from `mfs` to `lizardfs` (see second commit), as how would that help considering calling, for example mfssetgoal, would not replace anything in this case and would call mfssetgoal again, causing an infinite bash recursion (hence the error message). Seems like someone in the past did find and replace but didn't check if this would break the intended behavior here.

The CMake file that generates all the sfs* tools: https://github.com/leil-io/saunafs/blob/main/src/tools/CMakeLists.txt

Notice that all it does is create links for the sfs* tools to this script file. The saunafs binary doesn't touch this script at all. This is the same for LizardFS with mfs*.

https://github.com/lizardfs/lizardfs/blob/master/src/tools/CMakeLists.txt

> Are you sure that you have no more references to mfs in the code base (which caused issues in LizardFS)?

Positive

![image](https://github.com/user-attachments/assets/a77ed8b4-f9c5-4ff3-9334-d2ae6dbfcb80)

And neither LizardFS references either

![image](https://github.com/user-attachments/assets/636250a4-4015-4c5f-868c-462e66ae8bab)

Reply:
In fact, now that I think about it, this mfs to lizardfs change was never released in stable (since they stopped I think around 2018 making any new stable releases, and the commit date for the change was 2019), so this technically is a regression from LizardFS.

Reply:
Either way, thanks for taking the time to fix this. We are going to probably remove the sfs* links in 5.0.0 or later and we encourage using saunafs instead. However, while it's still here, might as well fix it as some people coming from LizardFS are still used to it. I was going to fix it myself, but then I remembered there was a hanging PR for this in LizardFS, so I thought I would ask you.

I'll try this out tomorrow and if it works, I think we can merge it (after it passes CI, which might take some time).